pkgs/desktops/cinnamon/cinnamon-session/default.nix

```diff
@@ -0,0 +1,101 @@
+{ fetchFromGitHub
+, cinnamon-desktop
+, cinnamon-settings-daemon
+, dbus-glib
+, docbook_xsl
+, docbook_xml_dtd_412
+, glib
+, gsettings-desktop-schemas
+, gtk3
+, libcanberra
+, libxslt
+, makeWrapper
+, meson
+, ninja
+, pkgconfig
+, python3
+, stdenv
+, systemd
+, wrapGAppsHook
+, xapps
+, xmlto
+, xorg
+, cmake
+, libexecinfo
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cinnamon-session";
+  version = "4.4.0";
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "0hplck17rksfgqm2z58ajvz4p2m4zg6ksdpbc27ki20iv4fv620s";
+  };
+
+  /* Run-time dependency libelogind found: NO (tried pkgconfig and cmake)
+  Run-time dependency libsystemd-login found: NO (tried pkgconfig and cmake)
+  Run-time dependency libsystemd found: NO (tried pkgconfig and cmake) */
+
+  buildInputs = [
+    cinnamon-desktop
+    cinnamon-settings-daemon
+    dbus-glib
+    glib
+    gsettings-desktop-schemas
+    gtk3
+    libcanberra
+    libxslt
+    makeWrapper
+    xapps
+    xmlto
+    xorg.xtrans
+    xorg.libXtst
+    systemd
+  ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    wrapGAppsHook
+    cmake
+    libexecinfo
+    docbook_xsl
+    docbook_xml_dtd_412
+    python3
+    pkgconfig
+  ];
+
+  NIX_CFLAGS_COMPILE = "-I${glib.dev}/include/gio-unix-2.0"; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
+  mesonFlags = [ "-Dgconf=false" ];
+
+  postPatch = ''
+    chmod +x data/meson_install_schemas.py # patchShebangs requires executable file
+    patchShebangs data/meson_install_schemas.py
+
+    substituteInPlace cinnamon-session/meson.build \
+      --replace "# elogind" "dbus_glib"
+
+    sed -i "51i dbus_glib = dependency('dbus-glib-1')" meson.build
+  '';
+
+  postFixUp = ''
+    for f in "$out/bin/"*; do
+      wrapProgram "$f" \
```

Comment:
we already have `wrapGAppsHook`, so already wrap applications.

Try extending `gappsWrapperArgs` to avoid wrapping twice.